izpack-compiler/src/main/java/com/izforge/izpack/compiler/packager/impl/PackagerBase.java

```diff
@@ -381,6 +384,41 @@ protected boolean packSeparateJars()
         return info != null && info.getWebDirURL() != null;
     }
 
+    private List<DynamicVariable> buildVariableList()
+    {
+        DirectedGraph<DynamicVariable> graph = new DirectedGraph<DynamicVariable>();
+        DynamicVariable rootVar = new DynamicVariableImpl("", "");
+
+        for (List<DynamicVariable> dynVariables : dynamicVariables.values())
+        {
+            for (DynamicVariable var : dynVariables)
+            {
+                graph.addEdge(rootVar, var);
+                for (String childName : var.getUnresolvedVariableNames())
+                {
+                    List<DynamicVariable> childVars = dynamicVariables.get(childName);
+                    if (childVars != null)
+                    {
+                        for (DynamicVariable childVar : childVars)
+                        {
+                            graph.addEdge(var, childVar);
+                        }
+                    }
+                }
+            }
+        }
+
+        Iterator<DynamicVariable> it = new BreadthFirstIterator<DynamicVariable>(graph, rootVar);
+        List<DynamicVariable> nodes = new ArrayList<DynamicVariable>();
+        while (it.hasNext()) {
+            nodes.add(it.next());
+        }
+        nodes.remove(rootVar);
```

Comment:
Ich werde ab  25.05.2016 nicht im B�ro sein. Ich kehre zur�ck am
30.05.2016.
Ich werde Ihre Nachricht nach meiner R�ckkehr beantworten.

Die R�ckmeldung bezieht sich auf ein Mail mit folgendem Thema:
Re: [izpack/izpack] IZPACK-1406:  Preorder dynamic variables according to
references among them at compile time (#509)

____________________________________________________________________________________________
Gesendet (c) GRZ/RACON Linz 2016 Agent 'Abwesenheit'

----------------------------------------
Raiffeisen Software GmbH
Firmenbuchnummer: 86804d, Landesgericht Linz
DVR: 0572934, UID: ATU 36961105

Der Austausch von Nachrichten mit dem oben angef�hrten Absender via E-Mail
dient ausschlie�lich Informationszwecken. Rechtsgesch�ftliche Erkl�rungen
d�rfen �ber dieses Medium nicht ausgetauscht werden.
Correspondence with above mentioned sender via e-mail is only for
information purposes. This medium is not to be used for the exchange of
legally-binding communications.
----------------------------------------